### DISCUSSION TITLE
Some Sql statement performance problems

### DISCUSSION THREAD

**Discussion by saki-is-username** [ory/hydra#3070] [Q&A] [answered]
When using Hydra, I found that when there is too much data in HYDRA_Oauth2_consent_request_handled table, the "join" in the relevant SQL statement of the HYDRa code results in low efficiency, that the SQL execution takes 2 to 3 seconds. I was wondering if this was considered previously?

**Comment by vinckr** [comment, marked as answer]
There is a cleanup routine that could help with that: https://www.ory.sh/docs/hydra/cli/hydra-janitor .

**Reply by saki-is-username** [reply]
I've already deployed Hydra using K8S, and it seems like the only way to execute this is through a script, which seems inconvenient

**Reply by Demonsthere** [reply]
Hi there, 
If this is a periodic operation, it could be automated as a CronJob in k8s and set to run at a certain timeframe :) We do not have currently this feature in the official charts, but I would gladly accept it a PR 😉

**Reply by saki-is-username** [reply]
If so it would be great

**Comment by aeneasr** [comment]
We’re also refactoring the storage layer in Hydra 2.0 which will have less JOINs and better performance!
